dev-tools/cookbook/RemoveCommentsFixer.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+// src/Fixer/Comment/RemoveCommentsFixer.php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\Comment;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author Your name <your@email.com>
+ */
+final class RemoveCommentsFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition(): FixerDefinition
+    {
+        // Return a definition of the fixer, it will be used in the documentation.
+        return new FixerDefinition(
+            'Removes all comments of the code that are preceded by `;` (semicolon).', // Trailing dot is important. We thrive to use English grammar properly.
+            [
+                new CodeSample(
+                    "<?php echo 123; /* Comment */\n"
+                ),
+            ]
+        );
+    }
+
+    // ...
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens): bool
+    {
+        // Check whether the collection is a candidate for fixing.
+        // Has to be ultra cheap to execute.
+        return $tokens->isTokenKindFound(T_COMMENT);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens): void
+    {
+        // Add the fixing logic of the fixer here.
+        foreach ($tokens as $index => $token) {
+            if (!$token->isGivenKind(T_COMMENT)) {
+                continue;
+            }
+
+            // need to figure out what to do here!
+            $prevTokenIndex = $tokens->getPrevMeaningfulToken($index);
+            $prevToken = $tokens[$prevTokenIndex];
+
+            if ($prevToken->equals(';')) {
+                $tokens->clearAt($index);
+            }
+        }
+    }
+
+    private function applyFixNoAction(): void
```

Comment:
where is this one coming from?

Reply:
This is for that one case in the cookbook (one with `// no action`), and all is removed in a way that `protected function applyFix(\SplFileInfo $file, Tokens $tokens): void` is right above that comment.

Do you think it would be better to have this comment at the end of `applyFix`, after the `foreach` loop?